data/transactions/logic/assembler.go

```diff
@@ -903,14 +907,31 @@ func asmBranch(ops *OpStream, spec *OpSpec, args []string) error {
 		return ops.error("branch operation needs label argument")
 	}
 
-	ops.referToLabel(ops.pending.Len(), args[0])
+	ops.referToLabel(ops.pending.Len()+1, args[0], ops.pending.Len()+spec.Size)
 	ops.pending.WriteByte(spec.Opcode)
 	// zero bytes will get replaced with actual offset in resolveLabels()
 	ops.pending.WriteByte(0)
 	ops.pending.WriteByte(0)
 	return nil
 }
 
+func asmSwitch(ops *OpStream, spec *OpSpec, args []string) error {
+	numOffsets := len(args)
+	if numOffsets > math.MaxUint8 {
+		return ops.errorf("%s cannot take more than 255 labels", spec.Name)
```

Comment:
Why not switch up to `256` labels here, i.e. switch from label0 to label255?

Reply:
Because we  encode the number of labels in the instruction bytecode with a byte. So a 0 in the byte code means _no_ labels, which is admittedly a little odd.  `switch` with no labels is basically equivalent to `pop` of an int.

The alternative would be to have the meaning be that a `n` byte in the bytecode means that there are `n+1` labels to come.  That seemed a little too cute.

So, short answer is: We encode the number of labels in a byte, and a byte only goes up to 255 (without playing games).

Reply:
By the way, if you *are* using switch on an int that came from a single byte, you can still precisely deal with each value. You use a `switch` with 255 labels (which deals with cases 0 through 254), and the fall-through handles the last case.

Reply:
> So a 0 in the byte code means no labels, which is admittedly a little odd. 

Ah got it, thanks for the explanation, resolving